### PR TITLE
589 Add backup flow after account creation

### LINKED ISSUE #589: [FEATURE] Add backup flow after account creation
# Issue to be solved

When a user creates a new profile in Flow Wallet mobile, after the profile is created, we must immediately prompt the user to optionally back up their account. This prompt must send the user down two paths, either to our existing backup view, or to the wallet dashboard.

The prompt can contain copy such as (suggestion): "Congratulations! Your Flow account is now ready. To best secure your account, you can backup your account now. " with buttons: "Back up" or "Maybe Later".

# Resource
https://www.figma.com/design/mSr5uIApRJNrpkmN0YNuD6/FLOW-REF-Mobile-Design-v2.0.0?node-id=25065-11991&t=3nqDGAO4U00gbe77-1

Image

### PR DESCRIPTION
**Pull request by jeden**
## Related Issue
- https://github.com/Outblock/FRW-iOS/issues/589

## Summary of Changes
Modified `CreateProfileWaitingView` to show the two buttons, and handled the added route in sequence after showing the home screen.

## Need Regression Testing

- [ ] Yes
- [x] No

## Risk Assessment

- [x] Low
- [ ] Medium
- [ ] High

## Screenshots (if applicable)
image

Before:

https://github.com/user-attachments/assets/c96b1315-842f-4c47-9658-16c90ee7ef35

After:

https://github.com/user-attachments/assets/3f0b9b96-1ca5-4b0a-97b4-ba6c9d6bc60b

### CODE REVIEW

File: FRW/Tools/Threading.swift
Comment (zhouxl) [inline FRW/Tools/Threading.swift:1]: `DispatchQueue.swift`

File: FRW/Modules/Register/ViewModel/UsernameViewModel.swift [hunk 1]
```diff
@@ -99,12 +99,16 @@ class UsernameViewModel: ViewModel {
...
             do {
                 let txid = try await UserManager.shared.register(currentText, mnemonic: nil)
-                let viewModel = CreateProfileWaitingViewModel(txId: txid ?? "") { _ in
+                let viewModel = CreateProfileWaitingViewModel(txId: txid ?? "") { _, createBackup in
 
                     DispatchQueue.main.async {
                         self.changeBackupTypeIfNeeded()
                         self.state.isRegisting = false
                         Router.popToRoot()
+
```
Comment (zhouxl) [inline FRW/Modules/Register/ViewModel/UsernameViewModel.swift:107]: Might need to add a delay?

Comment (jeden) [reply FRW/Modules/Register/ViewModel/UsernameViewModel.swift:107]: I thought about that, but it worked without any issue in my testing. Maybe I can test on a couple more devices and iOS versions, just to be sure

**Comment by lmcmz** [comment]
It feels wired to jump back to root can then instantly go to backup page.
Can we change the navigation behaviour?

**Comment by jeden** [comment]
> It feels wired to jump back to root can then instantly go to backup page. Can we change the navigation behaviour?

THe problem with this is that when the backup screen is dismissed, it will get back to the account creation screen, and from that point on it should redirect to home. I'll check if that can be done without any weird transition

**Comment by jeden** [comment]
@lmcmz How about this?
 I'm setting the window alpha to zero, popping to root, showing the backup screen, and animating the alpha back to 1 with a delay of 0.1 and duration of 0.4 secs

https://github.com/user-attachments/assets/fa484cc1-1c7a-418f-94f0-e0b7c2606fd4

**Comment by JeffreyDoyle** [comment]
Hey @jeden -- Let's please ensure there is no "sliding" animation shown. The Backup view needs to be immediately displayed once the user selects to Backup

**Comment by JeffreyDoyle** [comment]
Can we also change the copy on the final account creation view to "Congratulations! Your Flow account is now ready. To best secure your account, you can back it up now."

**Comment by lmcmz** [comment]
We can override the navigation stack to avoid user go to back the account creation page.

File: FRW/Services/Router/Router.swift [hunk 2]
```diff
@@ -47,6 +47,35 @@ extension Router {
         }
     }
 
+    static func popToRootAndRoute(to target: RouterTarget, animated: Bool = true) {
```
Comment (lmcmz) [inline FRW/Services/Router/Router.swift]: Please use `setViewControllers` to reset the navigation stack. 
 You can refer to
 https://github.com/Outblock/FRW-iOS/blob/86bdf99ef3e67141d67316c838ddbdec09815c56/FRW/Services/Router/RouteMap.swift#L142-L149

**Review by lmcmz — CHANGES_REQUESTED** [review]

**Review by zhouxl — APPROVED** [review]

**Review by lmcmz — APPROVED** [review]

**Comment by jeden** [comment]
@Peppermint1020 This is on Testflight version 2.3.0 build 51

**Pull request merged**